### DISCUSSION TITLE
Running in debug mode in PyCharm

### DISCUSSION THREAD

**Discussion by trax72** [ManimCommunity/manim#2498] [Q&A] [answered]
I'm trying to run manim in debug mode in PyCharm.

I read the discussion [here](https://github.com/3b1b/manim/issues/894) and [here](https://p14jeffwest.blogspot.com/2020/04/how-to-debug-manim-code-with-pycham.html) and tried to adapt it to Manim **community** version but I get the following errors:

```
C:\Users\User\Python3.9\Scripts\python.exe C:\Users\User\AppData\Roaming\JetBrains\IntelliJIdea2021.3\plugins\python\helpers\pydev\pydevd.py --multiproc --qt-support=auto --client 127.0.0.1 --port 52969 --file C:/Users/User/Python3.9/Lib/site-packages/manim ./Algorithm.py MovingVertices
Connected to pydev debugger (build 213.5744.223)
Traceback (most recent call last):
  File "C:\Users\User\AppData\Roaming\JetBrains\IntelliJIdea2021.3\plugins\python\helpers\pydev\pydevd.py", line 1483, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Users\User\AppData\Roaming\JetBrains\IntelliJIdea2021.3\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:/Users/User/Python3.9/Lib/site-packages/manim\__main__.py", line 6, in <module>
    from . import __version__, console
ImportError: attempted relative import with no known parent package
python-BaseException
```

In fact the files `__version__` and `console` don't exist in the manim directory at all and the following code lines also refer to various packages that do not exist there (at least not as it appears in the code), such as: `from .cli.cfg.group import cfg` etc.

Any advice would be appreciated...

Thanks,
Tal.

**Comment by behackl** [comment, marked as answer]
I find it easiest to debug animations by running them as pure Python files. That is, I have a file like

```py
from manim import *

class SquareToCircle(Scene):
    def construct(self):
        s = Square()
        c = Circle()
        self.add(s)
        self.play(Transform(s, c))


with tempconfig({"quality": "medium_quality"}):
    scene = SquareToCircle()
    scene.render()
```

and then I use the default Python debugger on it. Make sure the directory containing your code does not contain a directory "manim", otherwise things will break. Personally, I put my files in the example_scenes directory before running them.

**Reply by trax72** [reply]
Awesome, this is exactly what I was looking for, thanks!
